ledger/tracker.go

```diff
@@ -98,19 +116,73 @@ type ledgerForTracker interface {
 	GenesisHash() crypto.Digest
 	GenesisProto() config.ConsensusParams
 	GenesisAccounts() map[basics.Address]basics.AccountData
+
+	// TODO: temporary?
+	scheduleCommit(basics.Round)
+	waitAccountsWriting()
 }
 
 type trackerRegistry struct {
 	trackers []ledgerTracker
+	driver   *accountUpdates
+
+	// ctx is the context for the committing go-routine. It's also used as the "parent" of the catchpoint generation operation.
+	ctx context.Context
+	// ctxCancel is the canceling function for canceling the committing go-routine ( i.e. signaling the committing go-routine that it's time to abort )
+	ctxCancel context.CancelFunc
+	// committedOffset is the offset at which we'd like to persist all the previous account information to disk.
+	committedOffset chan deferredCommit
+	// commitSyncerClosed is the blocking channel for synchronizing closing the commitSyncer goroutine. Once it's closed, the
+	// commitSyncer can be assumed to have aborted.
+	commitSyncerClosed chan struct{}
+
+	// accountsWriting provides synchronization around the background writing of account balances.
+	accountsWriting sync.WaitGroup
+
+	// dbRound is always exactly accountsRound(),
+	// cached to avoid SQL queries.
+	dbRound basics.Round
+
+	dbs db.Pair
+	log logging.Logger
+
+	mu deadlock.RWMutex
+}
+
+func (tr *trackerRegistry) initialize(au *accountUpdates, l ledgerForTracker) (err error) {
+	tr.driver = au
+
+	tr.dbs = l.trackerDB()
+	tr.log = l.trackerLog()
+
+	err = tr.dbs.Rdb.Atomic(func(ctx context.Context, tx *sql.Tx) (err error) {
+		tr.dbRound, err = accountsRound(tx)
+		return err
+	})
+
+	if err != nil {
+		return err
+	}
+
+	tr.ctx, tr.ctxCancel = context.WithCancel(context.Background())
+	tr.committedOffset = make(chan deferredCommit, 1)
+	tr.commitSyncerClosed = make(chan struct{})
+	go tr.commitSyncer(tr.committedOffset)
+
+	return
 }
 
 func (tr *trackerRegistry) register(lt ledgerTracker) {
```

Comment:
the `register` can be combined with the `initialize` method. In particular, you can just pass in a slice of `ledgerTracker` object pointers.